libfoundation/src/foundation-string.cpp

```diff
@@ -1065,7 +1065,8 @@ uindex_t MCStringGetLength(MCStringRef self)
 const unichar_t *MCStringGetCharPtr(MCStringRef self)
 {
     if (__MCStringIsIndirect(self))
-        __MCStringResolveIndirect(self);
+		if (!__MCStringResolveIndirect(self))
```

Comment:
The GetCharPtr() and GetNativeCharPtr() method present a slight problem as posed. They are not meant to throw, instead returning nil if they can't satisfy the request.

However, __MCStringResolveIndirect(), and I'm guessing __MCStringUnnativize() might both throw out of memory.

The GetNativeCharPtr() call is pretty easy to fix - if indirect, and if indirect string is native then return it's native char buffer (we will need to check that the return pointer from GetNativeCharPtr() is not used across any calls to MCString functions).

The GetCharPtr() one is slightly more tricky - I think we currently assume that it always returns a non-nil value... But due to the potential for unnativization (depending on how that's implemented), it might fail.